pywren/executor.py

```diff
@@ -181,19 +167,16 @@ def map(self, func, iterdata, extra_env = None, extra_meta = None,
         func_str = func_and_data_ser[0]
         data_strs = func_and_data_ser[1:]
         data_size_bytes = sum(len(x) for x in data_strs)
-        s3_agg_data_key = None
-        host_job_meta['aggregated_data_in_s3'] = False
+        agg_data_key = None
+        host_job_meta['agg_data'] = False
         host_job_meta['data_size_bytes'] =  data_size_bytes
 
         if data_size_bytes < wrenconfig.MAX_AGG_DATA_SIZE and data_all_as_one:
-            s3_agg_data_key = s3util.create_agg_data_key(self.s3_bucket,
-                                                         self.s3_prefix, callset_id)
+            agg_data_key = self.storage.create_agg_data_key(callset_id)
```

Comment:
so one more thought -- all these utility functions to create keys, can we stick them into a `StorageUtils` or something like that ? These functions shouldn't change based on storage backend, is that right ?